fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedMetadata.java

```diff
@@ -0,0 +1,213 @@
+        ConnectorMetadata metadata = metadataOfTable(databaseName, tableName);
+        return metadata.listPartitionNames(databaseName, tableName);
+    }
+
+    @Override
+    public List<String> listPartitionNamesByValue(String databaseName, String tableName,
+                                                  List<Optional<String>> partitionValues) {
+        ConnectorMetadata metadata = metadataOfTable(databaseName, tableName);
+        return metadata.listPartitionNamesByValue(databaseName, tableName, partitionValues);
+    }
+
+    @Override
+    public Table getTable(String dbName, String tblName) {
+        ConnectorMetadata metadata = metadataOfTable(dbName, tblName);
+        return metadata.getTable(dbName, tblName);
+    }
+
+    @Override
+    public Pair<Table, MaterializedIndexMeta> getMaterializedViewIndex(String dbName, String tblName) {
+        ConnectorMetadata metadata = metadataOfTable(dbName, tblName);
+        return metadata.getMaterializedViewIndex(dbName, tblName);
+    }
+
+    @Override
+    public List<RemoteFileInfo> getRemoteFileInfos(Table table, List<PartitionKey> partitionKeys, long snapshotId,
+                                                   ScalarOperator predicate, List<String> fieldNames) {
+        ConnectorMetadata metadata = metadataOfTable(table);
+        return metadata.getRemoteFileInfos(table, partitionKeys, snapshotId, predicate, fieldNames);
+    }
+
+    @Override
+    public List<PartitionInfo> getPartitions(Table table, List<String> partitionNames) {
+        ConnectorMetadata metadata = metadataOfTable(table);
+        return metadata.getPartitions(table, partitionNames);
+    }
+
+    @Override
+    public Statistics getTableStatistics(OptimizerContext session, Table table, Map<ColumnRefOperator, Column> columns,
+                                         List<PartitionKey> partitionKeys, ScalarOperator predicate) {
+        ConnectorMetadata metadata = metadataOfTable(table);
+        return metadata.getTableStatistics(session, table, columns, partitionKeys, predicate);
+    }
+
+    @Override
+    public void clear() {
+        metadataMap.forEach((k, v) -> v.clear());
+    }
+
+    @Override
+    public void refreshTable(String srDbName, Table table, List<String> partitionNames, boolean onlyCachedPartitions) {
+        ConnectorMetadata metadata = metadataOfTable(table);
+        metadata.refreshTable(srDbName, table, partitionNames, onlyCachedPartitions);
+    }
+
+    @Override
+    public void createDb(String dbName) throws DdlException, AlreadyExistsException {
+        hiveMetadata.createDb(dbName);
+    }
+
+    @Override
+    public boolean dbExists(String dbName) {
+        return hiveMetadata.dbExists(dbName);
+    }
+
+    @Override
+    public void createDb(String dbName, Map<String, String> properties) throws DdlException, AlreadyExistsException {
+        hiveMetadata.createDb(dbName, properties);
```

Comment:
iceberg catalog can't create db by hms

Reply:
I think creating db is irrelevant to specifical catalog type